### ISSUE TITLE
Crash after 10s (network related - libnetwork)

### ISSUE THREAD

**Issue by larsklingsten** [Microsoft/vscode#24133]
- VSCode Version: Code 1.11.1 (d9484d12b38879b7f4cdd1150efeb2fd2c1fbf39, 2017-04-06T14:00:34.044Z)
- OS Version: Darwin x64 16.5.0
- Extensions: 

|Extension|Author|Version|
|---|---|---|
|dart-code|DanTup|1.1.0|
|python|donjayamanne|0.6.0|
|Angular2|johnpapa|2.2.3|
|Go|lukehoban|0.6.59|
|cpptools|ms-vscode|0.10.5|
|csharp|ms-vscode|1.8.1|;

---

Steps to Reproduce:

1. open "vs code"
2. wait 10-20 seconds
3. vs code crashed "Code quite unexpectedly" (or sometimes "Electron" instead of "code"=
[vs code - crash.txt](https://github.com/Microsoft/vscode/files/903778/vs.code.-.crash.txt)
(have tried to uninstall then re-install)

The program appears to work ok if the the wifi is turned off when vs code is started. Its can be turned on only after a popup says "warning you are not connected to the internet"

**Comment by chrmarti** [comment]
Does this still happen if you set `"git.enabled": false` in your settings?

**Comment by EreMaijala** [comment]
Possibly related: Same Mac OS version, 1.11.0 works fine but 1.11.1 hung after upgrade and won't start. I can see Code and Code Helper running, but no window is created. Killing the processes and trying again does not help. Downgrading to 1.11.0 makes everything work again.

**Comment by EreMaijala** [comment]
I also tried to re-download the full application, didn't help.

**Comment by chrmarti** [comment]
@EreMaijala That sounds like a different problem. Could you open a new issue for that? /fyi @bpasero

**Comment by EreMaijala** [comment]
@chrmarti Ok, sorry for trying to hijack this! Opened #24179.

**Comment by bpasero** [comment]
The relevant part of the crash seems to be this and looks to be network related:

```
Thread 1 Crashed:: Dispatch queue: com.apple.network.connections
0   libnetwork.dylib              	0x00007fffea2bfa19 nw_protocol_coretls_get_input_frames + 406
1   libnetwork.dylib              	0x00007fffea26fb01 nw_endpoint_flow_service_reads + 806
2   libnetwork.dylib              	0x00007fffea2749f5 nw_endpoint_handler_add_read_request + 675
3   libnetwork.dylib              	0x00007fffea2a8ec3 __nw_connection_read_block_invoke.136 + 38
4   libdispatch.dylib             	0x00007fffeabad524 _dispatch_call_block_and_release + 12
5   libdispatch.dylib             	0x00007fffeaba48fc _dispatch_client_callout + 8
6   libdispatch.dylib             	0x00007fffeabba9a0 _dispatch_queue_serial_drain + 896
7   libdispatch.dylib             	0x00007fffeabad306 _dispatch_queue_invoke + 1046
8   libdispatch.dylib             	0x00007fffeaba66b5 _dispatch_root_queue_drain + 476
9   libdispatch.dylib             	0x00007fffeaba648c _dispatch_worker_thread3 + 99
10  libsystem_pthread.dylib       	0x00007fffeadf3616 _pthread_wqthread + 1299
11  libsystem_pthread.dylib       	0x00007fffeadf30f1 start_wqthread + 13
```

@larsklingsten did you configure a proxy? is this working from a corp network or at home? does it make any difference if you run the application from the command line vs. dock?

**Comment by larsklingsten** [comment]
Please note my issue is covered twice https://github.com/Microsoft/vscode/issues/24136#issuecomment-292497839 (one without extensions and one without) - perhaps these can be merged.

1) crashes both when run from terminal and dock.
2) home network - no proxies
3) DCHP ip addresses with own setting of DNS server (switching all to auto - does not change anything)

**Comment by larsklingsten** [comment]
Dear bpasero, 
Thank you for you efforts - I am downgrading to osx 10.11.6 -> 10.12.3 also killed my golang debugging (delve), as believe that there are simply too many issues with 10.12.

**Comment by bpasero** [comment]
@larsklingsten can you check if the previous VS Code is also crashing for you? This would point into a direction that this might be OS update related and not VS Code. The previous version is at: https://az764295.vo.msecnd.net/stable/8076a19fdcab7e1fc1707952d652f0bb6c6db331/VSCode-darwin-stable.zip

**Comment by larsklingsten** [comment]
Hi Benjamin - i am affald i can't help on this matter - as i have
downgraded from macOS 10.12 to osx 10.11 -  Too much trouble with 10.12
also in respect of my golang debugging chain.  Lars

**Comment by bpasero** [comment]
Please reopen if this still reproduces with our upcoming 1.12 release.
